core/src/main/java/fr/sncf/osrd/train/TrainPositionTracker.java

```diff
@@ -86,15 +72,53 @@ public boolean equals(Object obj) {
             return false;
 
         var other = (TrainPositionTracker) obj;
-        return trackSectionRanges.equals(other.trackSectionRanges);
+        return trackSectionPath.equals(other.trackSectionPath)
+                && headPathPosition == other.headPathPosition
+                && tailPathPosition == other.tailPathPosition;
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(trackSectionRanges);
+        return Objects.hash(trackSectionPath, headPathPosition, tailPathPosition);
     }
     // endregion
 
+    /** Resets the position tracker, but keeps the pre-computed data */
+    public TrainPositionTracker reset() {
+        headPathPosition = 0;
+        tailPathPosition = 0;
+        return this;
```

Comment:
Does it make sens to have the head and the trail of the train both at zero ? Don't we want them spaced by the length of the train at any time ?

Reply:
I didn't change the behavior here, the train "appears" as it starts moving until it has reached its size. I could look into changing this behavior now that the class is simpler, but it would go into another branch